libs/api/Verifier.cpp

```diff
@@ -555,10 +608,29 @@ ebpf_api_elf_verify_section_from_memory(
     const char** error_message,
     ebpf_api_verifier_stats_t* stats)
 {
-    std::stringstream stream(std::string(data, data_length));
-    struct _thread_local_storage_cache tls_cache;
+    *error_message = nullptr;
+    *report = nullptr;
+    try {
+        if (!ElfCheckElf(
+                data_length,
+                const_cast<uint8_t*>(reinterpret_cast<const uint8_t*>(data)),
+                static_cast<uint32_t>(data_length))) {
+            throw std::runtime_error("ELF file is malformed: " + _elf_everparse_error);
+        }
 
-    set_global_program_and_attach_type(program_type, nullptr);
-    set_verification_in_progress(true);
-    return _ebpf_api_elf_verify_section_from_stream(stream, "memory", section, verbose, report, error_message, stats);
+        std::stringstream stream(std::string(data, data_length));
+        struct _thread_local_storage_cache tls_cache;
+
+        set_global_program_and_attach_type(program_type, nullptr);
```

Comment:
since this is same as parts of `ebpf_api_elf_verify_section_from_file` - can the try/catch block be changed to a inline function?

Reply:
I agree wholeheartedly. I think we should cleanup the entire file at some point. https://github.com/microsoft/ebpf-for-windows/issues/1234